core/src/main/scala/akka/kafka/javadsl/Transactional.scala

```diff
@@ -54,6 +55,31 @@ object Transactional {
       .map(_._1)
       .asJava
 
+  /**
+   * Internal API. Work in progress.
+   *
+   * The `partitionedSource` is a way to track automatic partition assignment from kafka.
+   * Each source is setup for for Exactly Only Once (EoS) kafka message semantics.
+   * To enable EoS it's necessary to use the [[Transactional.sink]] or [[Transactional.flow]] (for passthrough).
+   * When Kafka rebalances partitions, all sources complete before the remaining sources are issued again.
+   *
+   * By generating the `transactionalId` from the [[TopicPartition]], multiple instances of your application can run
+   * without having to manually assign partitions to each instance.
+   */
+  @ApiMayChange
+  @InternalApi
+  private[kafka] def partitionedSource[K, V](
```

Comment:
We should comment out the Java API. Java doesn't support scoped private visibility and it will seem public to Java users.